backbone.js

```diff
@@ -613,43 +629,48 @@
 
       options = _.extend({validate: true, parse: true}, options);
       var wait = options.wait;
+      var success = options.success || _.identity;
+      var error = options.error;
+      delete options.success;
+      delete options.error;
 
       // If we're not waiting and attributes exist, save acts as
       // `set(attr).save(null, opts)` with validation. Otherwise, check if
       // the model will be valid when the attributes, if any, are set.
       if (attrs && !wait) {
-        if (!this.set(attrs, options)) return false;
+        if (!this.set(attrs, options)) return Promise.reject(this);
```

Comment:
As I have defined it, Model#save will return a `Promise` that resolves to the model in the successful case. I am unclear that calling save on an unmodified model is a success case. If it isn't then the returned `Promise` should be rejected with the model (or maybe `false`?) If it is a success case then the returned `Promise` should be resolved with the model. Alternatively, the current behaviour of returning `false` could be preserved, in which case, someone like me would wrap every call to save like this:

    Promise.resolve(model.save(attrs)).then(function(saved) {
      if (saved === false) doSomethingSpecial(model);
      doSomethingExpected(saved || model);
    }, handleError);

The fourth option (as I see it) is to return `Promise.resolve(false)` which is kind of saying:
> "when save is successful, you'll get a resolved `Promise`, in the normal case it'll be fulfilled with the model and in the exceptional case it won't, it'll be fulfilled with `false`, otherwise (in the case of failures) the `Promise` will be rejected."